crux-core/src/crux/mem_kv.clj

```diff
@@ -85,31 +89,44 @@
     (log/debug "Using fsync on MemKv has no effect."))
 
   (count-keys [_]
-    (count @db))
+    (count @!db))
 
-  (db-dir [_]
-    (str db-dir))
+  (db-dir [_] nil)
 
   (kv-name [this]
     (.getName (class this)))
 
+  cp/CheckpointSource
+  (save-checkpoint [this dir]
+    (persist-db dir @!db)
+    {:tx (kvi/latest-completed-tx this)})
+
   Closeable
   (close [_]
-    (when db-dir
-      (persist-db db-dir db))))
-
-(defn ->kv-store {::sys/args {:db-dir {:required? false
-                                       :doc "Directory to (optionally) store K/V files"
-                                       :spec ::sys/path}
-                              :persist-on-close? {:required? true
-                                                  :default false
-                                                  :spec ::sys/boolean}}}
+    (cio/try-close cp-job)))
+
+(def ^:private cp-format
+  {:index-version c/index-version
+   ::version 1})
```

Comment:
Just wondering if this is correct, see the other KV stores have string versions?

Reply:
doesn't strictly matter so long as it's consistent for mem-kv - but yes, agreed, consistency's better :)